### DISCUSSION TITLE
Why not just use the system's own spell checker in macOS?

### DISCUSSION THREAD

**Discussion by WolfgangDpunkt** [Zettlr/Zettlr#3660] [Q&A]
I know the quality of the spell checker and possible alternatives has already been discussed. But a solution that seems obvious to me, I have not found mentioned so far: At least in macOS, surely one could just use the system's very good spell checker?
As a layman, it seems to me that this [only requires the activation of an option in the Electron creation.](https://www.electronjs.org/de/docs/latest/tutorial/spellchecker)
But probably it is more complicated?
(is there optionally the possibility to activate this on the user side in some config file?)

To me the spell checker would help a lot to catch typos immediately. Fiddling around with extra dictionaries etc. doesn't seem to me to be effective, as has already been discussed here.

Thanks!

**Comment by nathanlesage** [comment]
> Why not just use the system's own spell checker in macOS?

2 answers:

1. By disabling the current spellchecker we would save about 150MB of space in the binary (good) but would lose EVERY piece of customizability (very bad), and it's not guaranteed to work as expected at all
2. The macOS spellchecker would require you to switch the language in the system settings, rendering it very cumbersome to write multilingually

**Reply by julian-weinert** [reply]
This is incorrect. You can add and spell check multiple languages in macOS. Also the macOS spell checker is MUCH better than anything else, especially with multi language. Where others still try to parse what language this is about the system spell checker happily spell checked 150 sentences with mixed language within one sentence.
Using external spell checkers on mac instead of the system one makes absolutely no sense and is a complete UX disaster (spell check learns words across different programs)

**Reply by nathanlesage** [reply]
If anything in electrons implementation has changed in the past 3y, and if we can add this without jeopardizing the existing solutions, I'm happy to merge a PR!

**Comment by WolfgangDpunkt** [comment]
Dear Hendrik,
thank you for the quick reply and the explanations.

For me as a Mac user, it would already be helpful if this could be activated via an advanced setting. Even though it would be great to be able to save so much storage space in general, the fact that the old spell checker remains than still unused in the application code would not be so tragic. As long as it becomes possible that the editor window is "accessible" for the system spell checker process. That it would mean too much extra effort to provide two versions, once with and once without the old spell checker, is understandable.
So one could leave everything as it is for the standard user, but add a (possibly hidden) option for macOS to optionally switch to the system spell checker process.

**Comment by WolfgangDpunkt** [comment]
Hello again, I would like to politely push again on this issue. 
Since I now write notes almost daily with Zettlr (because I have really come to appreciate the application), the problem of the missing spell checker is becoming more and more pressing. Because I am relentlessly producing a lot of typos, all things I have to correct later.
Of course, one could write more carefully, but I fear the ubiquitous automated spell checking (Word, Firefox, Thunderbird, iOS....everywhere, almost everywhere...) has spoiled me.

In this respect, I would be very grateful for any way or workaround that Zettlr allows a spell checking on the part of macOS.
Because it's a shame, even though the brillant Duden dictionary is close at hand (I think that's what the German checking system from Apple relies on), to still produce a lot of text of poor quality, or to have to copy the texts to Word manualy.

Therefore, and if it would only be a hidden option, as I said, to prevent all disadvantages and risks for other user, I would really be super grateful for a solution.

**Comment by supermueller** [comment]
Is this problem still not solved? For me the most important argument is, that I use spell checking system wide and don’t want to maintain a second dictionary.

Mac users are used to using the same services across different applications, so not allowing them to do this, is a hard restriction.

@nathanlesage You don’t have to change the language in system preferences, because you can add several languages for spell checking (see https://support.apple.com/en-us/guide/mac-help/mchlp2299/mac).

**Reply by nathanlesage** [reply]
The issue is still that Electron has not yet implemented an API that gives me raw access to the macOS spellchecker. Until that is fixed with Electron, I'm not sure I can implement it. I have tried to implement the system spellcheck once in the past and couldn't because of that.

**Reply by supermueller** [reply]
I know nothing about Electron but their documentation says that it has an build in spellchecker that makes use of macOS’s native spellchecker API (https://www.electronjs.org/docs/latest/tutorial/spellchecker/)

**Reply by nathanlesage** [reply]
I am well aware of this API but it doesn't work in a way that I would need to integrate it.

**Comment by qingster2** [comment]
Similar boat here. I just started using Zettlr and the spellchecker is picking up a lot of words as wrong but they are actually correct (I write about medicine so there are words on anatomy disease molecules medications etc). Example ketamine is underlined red, I’m like seriously?? Are you using a spellchecker for 3rd graders?  These words are more often recognized as correct by apple’s native spellchecker and Google spellchecker (which I think is one of the best bc when I write in google docs it even auto capitalize brand drug names).  LanguageTool spellchecker is particularly bad about this too. Can we do something to improve? It’s tedious have to add “Ketamine” and “ketamine” and “ketamine’s” etc… to dictionary repeatedly.  And it’s big distraction looking at a page filled with red underlines.

**Reply by nathanlesage** [reply]
The technical reason why we don't implement electron's built-in spell checker (which would hook into the macOS spell checker) is because this does not allow us the amount of control we need. It's fine if you only have some plain text in a window, but we have to programmatically check words, and electron does not offer any API for that. If we were to implement it, it would mess with Markdown formatting, render the user dictionary useless, and would possibly even interfere with other places.
